src/documentation/runners/direct.md

```diff
@@ -1,9 +1,47 @@
 ---
 layout: default
-title: "Apache Direct Runner"
+title: "Direct Runner"
 permalink: /documentation/runners/direct/
 redirect_from: /learn/runners/direct/
 ---
 # Using the Direct Runner
 
-This page is under construction ([BEAM-505](https://issues.apache.org/jira/browse/BEAM-505)).
+The direct runner executes pipelines on your machine and is designed to validate that pipelines adhere to the Apache Beam model as closely as possible. Instead of focusing on efficient pipeline execution, the direct runner performs additional checks to ensure that users do not rely on semantics that are not guaranteed by the model. Some of these checks include:
+
+* enforcing immutability of elements
+* enforcing encodability of elements
+* elements are processed in an arbitrary order at all points
+* serialization of user Fns (DoFn, CombineFn, etc.)
+
+Using the direct runner for testing and development helps ensure that pipelines are robust across different Beam runners. In addition, debugging failed runs can be a non-trivial task when a pipeline executes on a remote cluster. Instead, it is often faster and simpler to perform local unit testing on your pipeline code. Unit testing your pipeline locally also allows you to use your preferred local debugging tools.
+
+Here are some resources with information about how to test your pipelines.
+* [Testing Unbounded Pipelines in Apache Beam](https://beam.incubator.apache.org/blog/2016/10/20/test-stream.html) talks about the use of Java classes [PAssert]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/testing/PAssert.html) and [TestStream]({{ site.baseurl }}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/testing/TestStream.html) to test your pipelines.
```

Comment:
Please don't hardcode the full URL -- it is fragile and guaranteed to break when we go TLP. Please link to `baseurl/blog...`.